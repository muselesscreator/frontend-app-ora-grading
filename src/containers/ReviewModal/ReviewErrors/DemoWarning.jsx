import React from 'react';

import { FormattedMessage } from '@edx/frontend-platform/i18n';

import messages from './messages';

import ReviewError from './ReviewError';

/**
 * <DemoErrors />
 */
export const DemoErrors = () => {
  if (!process.env.REACT_APP_NOT_ENABLED) { return null; }
  return (
    <ReviewError
      key="demoMode"
      headingMessage={messages.demoModeHeading}
      variant="warning"
    >
      <FormattedMessage {...messages.demoModeMessage} />
    </ReviewError>
  );
};

export default DemoErrors;
