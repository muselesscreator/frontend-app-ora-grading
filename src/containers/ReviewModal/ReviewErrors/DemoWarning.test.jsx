import React from 'react';
import { shallow } from 'enzyme';

import DemoWarning from './DemoWarning';

let el;

describe('DemoErrors component', () => {
  describe('snapshots', () => {
    const OLD_ENV = process.env;
    beforeAll(() => {
      jest.resetModules();
      process.env = { ...OLD_ENV };
    });
    afterAll(() => {
      process.env = OLD_ENV;
    });
    test('does not render if disabled flag is missing', () => {
      process.env.REACT_APP_NOT_ENABLED = false;
      el = shallow(<DemoWarning />);
      expect(el).toMatchSnapshot();
      expect(el.isEmptyRender()).toEqual(true);
    });
    test('snapshot: disabled flag is present', () => {
      process.env.REACT_APP_NOT_ENABLED = true;
      el = shallow(<DemoWarning />);
      expect(el).toMatchSnapshot();
      expect(el.isEmptyRender()).toEqual(false);
    });
  });
});
