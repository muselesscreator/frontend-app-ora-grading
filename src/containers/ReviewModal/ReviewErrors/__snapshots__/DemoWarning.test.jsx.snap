// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoErrors component snapshots does not render if disabled flag is missing 1`] = `""`;

exports[`DemoErrors component snapshots snapshot: disabled flag is present 1`] = `
<ReviewError
  actions={Object {}}
  headingMessage={
    Object {
      "defaultMessage": "Demo Mode",
      "description": "Demo mode heading",
      "id": "ora-grading.ReviewModal.demoHeading",
    }
  }
  key="demoMode"
  variant="warning"
>
  <FormattedMessage
    defaultMessage="You are using the Demo Mode of the new Enhanced ORA Staff Grader interface.  You will be unable to submit grades until you activate the feature."
    description="Demo mode message"
    id="ora-grading.ReviewModal.demoMessage"
  />
</ReviewError>
`;
